integrationTests/state/stateTrie/stateTrie_test.go

```diff
@@ -831,6 +833,229 @@ func TestAccountsDB_RevertDataStepByStepWithCommitsAccountDataShouldWork(t *test
 	require.Equal(t, hrRoot2, hrRoot2Rev)
 }
 
+func testMigrateCodeLeaf(
+	t *testing.T,
+	testContext *vm.VMTestContext,
+	rcvAddr []byte,
+) {
+	testContext.CleanIntermediateTransactions(t)
+
+	txData := []byte("MigrateCodeLeaf@aa@bb@00")
+	sndAddr := []byte("12345678901234567890123456789112")
+	scr := &smartContractResult.SmartContractResult{
+		Value:    big.NewInt(0),
+		RcvAddr:  rcvAddr,
+		SndAddr:  sndAddr,
+		Data:     txData,
+		GasLimit: 10000,
+		GasPrice: 10,
+		CallType: 1,
+	}
+	returnCode, errProcess := testContext.ScProcessor.ProcessSmartContractResult(scr)
+	require.Nil(t, errProcess)
+	require.Equal(t, vmcommon.Ok, returnCode)
+
+	intermediate := testContext.GetIntermediateTransactions(t)
+	require.Equal(t, 1, len(intermediate))
+}
+
+func TestAccountsDB_RevertDataStepByStepWithCommitsAccountDataWithMigratedCode(t *testing.T) {
+	t.Parallel()
+
+	t.Run("account with migrated code leaf activation, should save code data to storage directly", func(t *testing.T) {
+		t.Parallel()
+
+		enableEpochs := config.EnableEpochs{
+			MigrateCodeLeafEnableEpoch: 0,
+		}
+
+		shardCoordinator, _ := sharding.NewMultiShardCoordinator(3, 1)
+		gasScheduleNotifier := vm.CreateMockGasScheduleNotifier()
+
+		storer := integrationTests.CreateMemUnit()
+		trieStorage, _ := integrationTests.CreateTrieStorageManager(storer)
+
+		testContext, err := vm.CreatePreparedTxProcessorWithVMsWithShardCoordinatorDBAndGas(enableEpochs, shardCoordinator, storer, gasScheduleNotifier)
+		require.Nil(t, err)
+		defer testContext.Close()
+
+		key := []byte("ABC")
+		val := []byte("123")
+		newVal := []byte("124")
+		adr := integrationTests.CreateRandomAddress()
+		adr[31] = 1 // set address for shard 1
+
+		adb := testContext.Accounts
+
+		stateMock, err := adb.LoadAccount(adr)
+		require.Nil(t, err)
+		_ = stateMock.(state.UserAccountHandler).SaveKeyValue(key, val)
+
+		err = adb.SaveAccount(stateMock)
+		require.Nil(t, err)
+		rootHash, err := adb.RootHash()
```

Comment:
Remove the rootHash checks and saveKeyValue inserts that are not needed to verify the code migration. Also valid for the next test.

Reply:
remove unnecessary code from integration tests